src/Microsoft.DotNet.Cli.Utils/EmbedAppNameInHost.cs

```diff
@@ -0,0 +1,155 @@
+            {
+                Directory.CreateDirectory(destinationDirectory);
+            }
+
+            // Copy AppHostSourcePath to ModifiedAppHostPath so it inherits the same attributes\permissions.
+            File.Copy(appHostSourcePath, ModifiedAppHostPath);
+
+            // Re-write ModifiedAppHostPath with the proper contents.
+            using (FileStream fs = new FileStream(ModifiedAppHostPath, FileMode.Truncate, FileAccess.ReadWrite, FileShare.Read))
+            {
+                fs.Write(array, 0, array.Length);
+            }
+
+            return ModifiedAppHostPath;
+        }
+
+        // See: https://en.wikipedia.org/wiki/Knuth%E2%80%93Morris%E2%80%93Pratt_algorithm
+        private static int[] ComputeKMPFailureFunction(byte[] pattern)
+        {
+            int[] table = new int[pattern.Length];
+            if (pattern.Length >= 1)
+            {
+                table[0] = -1;
+            }
+            if (pattern.Length >= 2)
+            {
+                table[1] = 0;
+            }
+
+            int pos = 2;
+            int cnd = 0;
+            while (pos < pattern.Length)
+            {
+                if (pattern[pos - 1] == pattern[cnd])
+                {
+                    table[pos] = cnd + 1;
+                    cnd++;
+                    pos++;
+                }
+                else if (cnd > 0)
+                {
+                    cnd = table[cnd];
+                }
+                else
+                {
+                    table[pos] = 0;
+                    pos++;
+                }
+            }
+            return table;
+        }
+
+        // See: https://en.wikipedia.org/wiki/Knuth%E2%80%93Morris%E2%80%93Pratt_algorithm
+        private static int KMPSearch(byte[] pattern, byte[] bytes)
+        {
+            int m = 0;
+            int i = 0;
+            int[] table = ComputeKMPFailureFunction(pattern);
+
+            while (m + i < bytes.Length)
+            {
+                if (pattern[i] == bytes[m + i])
+                {
+                    if (i == pattern.Length - 1)
+                    {
+                        return m;
+                    }
+                    i++;
+                }
+                else
+                {
+                    if (table[i] > -1)
+                    {
+                        m = m + i - table[i];
+                        i = table[i];
+                    }
+                    else
+                    {
+                        m++;
+                        i = 0;
+                    }
+                }
+            }
+            return -1;
+        }
+
+        private static void SearchAndReplace(byte[] array, byte[] searchPattern, byte[] patternToReplace, string appHostSourcePath)
+        {
+            int offset = KMPSearch(searchPattern, array);
+            if (offset < 0)
+            {
+                throw new Exception(appHostSourcePath + placeHolder);
```

Comment:
Never throw System.Exception.

Also, this should be a proper error message. I assume this is just unfinished fixup from pulling the code from SDK?

Reply:
Agree. It is a temp fix. I should make it a proper exception